src/vma/dev/ring_bond.cpp

```diff
@@ -53,15 +53,21 @@
 #undef  MODULE_HDR
 #define MODULE_HDR	 	MODULE_NAME "%d:%s() "
 
+/* Set limitation for number of rings for bonding device */
+#define MAX_NUM_RING_RESOURCES 10
+
+
 ring_bond::ring_bond(int count, net_device_val::bond_type type, net_device_val::bond_xmit_hash_policy bond_xmit_hash_policy, uint32_t mtu) :
 ring(count, mtu), m_lock_ring_rx("ring_bond:lock_rx"), m_lock_ring_tx("ring_bond:lock_tx") {
+	if (m_n_num_resources > MAX_NUM_RING_RESOURCES) {
+		ring_logpanic("Error creating bond ring with more than %d resource", MAX_NUM_RING_RESOURCES);
```

Comment:
@igor-ivanov logpanic should only be used for unexpected SW flows. In this case the number of rings per bond is user configuration parameter and as such we should only alert the user, but not crashing the system.